RecoLocalTracker/SiPixelRecHits/src/SiPixelTemplateReco2D.cc

```diff
@@ -531,9 +547,11 @@ int SiPixelTemplateReco2D::PixelTempReco2D(int id, float cotalpha, float cotbeta
             ystep = minv12*sumptdt1 + minv22*sumptdt2;
 
          } else {
+
 //  Assume alternately that ystep = 0 and then xstep = 0
-            xstep = sumptdt1/sumdtdt11;
-            ystep = sumptdt2/sumdtdt22;  
+
+            if(fabs(sumdtdt11) > 0.0001f) {xstep = sumptdt1/sumdtdt11;} else {xstep = 0.f;}

```

Comment:
```sumdtdt11``` and ```sumdtdt22``` are non negative by construction: "fabs" not needed here:
```C++
            xstep = (sumdtdt11 > 0.0001f ? sumptdt1/sumdtdt11 : 0.f);
            ystep = (sumdtdt22 > 0.0001f ? sumptdt2/sumdtdt22 : 0.f);
```

You may think about also replacing ```fabs``` with ```std::abs``` everywhere in this class.

Reply:
Also ```D``` at L532 above is non negative by definition

Reply:
We will remove the fabs conditional on the sumdtd11 and sumdtd22 because they are non-negative by construction. 
The D is a difference of two positive numbers, and could possibly be negative in some pathological case. So we would like to keep the conditional on 532

Reply:
ok